sunpy/image/slit.py

```diff
@@ -0,0 +1,171 @@
+import numpy as np
+
+import astropy.units as u
+
+
+def slit(mcube_in, range_a, range_b, N_slits=0, shift_x=+1, shift_y=-1,):
+    """
+    Returns an array with intensity along the slit on the y axis and time
+    along x.
+
+    Parameters
+    ----------
+    mcube_in : `sunpy.map.MapCube`
+        A mapcube of the images you want to perform the slit analysis on.
+        Usually with x and y as space, and z as time.
+        The x and y axes must be equal for all instances within the mapcube.
+    range_a : `astropy.units.Quantity`
+        A list of two `astropy.unit.Quantity` objects representing x1 and x2,
+        start and end of slit in x.
+    range_b : `astropy.units.Quantity`
+        A list of two `astropy.unit.Quantity` objects representing x2 and y2,
+        start and end of slit in y.
+    shift_x : `int`
+        Possible values, `-1`, `0`, `+1`
+        The displacement from the origin on the slit in x. These extra slits
+        calcuate a mean along each row to make the positioning of the original
+        slit less sensetive. Both `x` and `y` cannot be `0`. Default = `+1`
+    shift_y : `int`
+        Possible values, `-1`, `0`, `+1`
+        The displacement from the origin in `y`. Default = `-1`
+    N_slits : `int`
```

Comment:
This should be above `shift_x` so that the docs and kwargs are in the same order.

Also I don't find this very clear, something like "Number of extra pixels to extract on either side of the slit", which is also not very good.